test/prepare_disk.sh

```diff
@@ -0,0 +1,431 @@
+#!/bin/sh
+#
+# Copyright (C) 2018 Ultimaker B.V.
+# Copyright (C) 2018 Raymond Siudak <raysiudak@gmail.com>
+# Copyright (C) 2018 Olliver Schinagl <oliver@schinagl.nl>
+#
+# SPDX-License-Identifier: AGPL-3.0+
+
+set -eu
+
+ARM_EMU_BIN="${ARM_EMU_BIN:-}"
+
+CWD="$(pwd)"
+
+SYSTEM_UPDATE_DIR="${SYSTEM_UPDATE_DIR:-/etc/system_update}"
+PARTITION_TABLE_FILE="test_jedi_emmc_sfdisk.table"
+TARGET_STORAGE_DEVICE=""
+
+JEDI_PARTITION_TABLE_FILE_NAME="config/jedi_emmc_sfdisk.table"
+PREPARE_DISK_COMMAND="/etc/system_update.d/30_prepare_disk.sh"
+
+STORAGE_DEVICE_IMG="storage_device.img"
+MIN_PARTITION_SIZE="78124"    # 40 MiB (enough for f2fs and ext4)
+BYTES_PER_SECTOR="512"
+STORAGE_DEVICE_SIZE="7553024" # sectors, about 3.6 GiB
+BOOT_START="2048"             # offset 1 MiB
+ROOTFS_START="67584"          # offset 33 MiB
+USERDATA_START="1998848"      # offset 976 MiB
+
+TEST_OUTPUT_FILE="$(mktemp -d)/test_results_$(basename "${0%.sh}").txt"
+
+toolbox_image=""
+toolbox_root_dir=""
+work_dir=""
+
+exit_on_failure=false
+result=0
+
+# shellcheck source=test/include/armhf_emulation.sh
+. "test/include/armhf_emulation.sh"
+
+
+execute_prepare_disk()
```

Comment:
i assume everything below is just a copy/paste from the other script, so ignoring that

Reply:
Yes except for some variable naming and the location of the temporary files.